### PR TITLE
HTCONDOR-2254 Improve eventlog reader reliability

### PR DESCRIPTION
**Pull request by johnkn**
by having it always read lines and have less need to seek to recover from short reads.

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_utils/condor_event.cpp [hunk 1]
```diff
@@ -267,14 +292,38 @@ ULogEvent::~ULogEvent (void)
 {
 }
 
+/*static*/
+ULogEventNumber ULogEvent::readEventNumber(ULogFile& ulf, char* headbuf, size_t bufsize)
+{
+	ASSERT(bufsize > 32); // header is something like: 000 (16091.000.000) 01/22 12:09:19
+	memset(headbuf, 0, 32);
```
Comment (ColeBollig) [inline src/condor_utils/condor_event.cpp:299]: Is memset() supposed to be making headbuf with size 32 or is it supposed to be bufsize?

File: src/condor_utils/condor_event.cpp [hunk 2]
```diff
@@ -331,6 +380,111 @@ const char* ULogEvent::eventName(void) const
...
+		// now scan for the space after the time, so we can return that
+		datend = strchr(datend+1, ' ');
+	} else {
+		// here we expect the date and time to be in iso8601 format
+		// this will set -1 into all fields of dt that are not set by date/time parsing code
+		iso8601_to_time(p, &dt, &event_usec, &is_utc);
+	}
+
+	// check for a bogus date stamp
+	if (dt.tm_mon < 0 || dt.tm_mon > 11 || dt.tm_mday < 0 || dt.tm_mday > 32 || dt.tm_hour < 0 || dt.tm_hour > 24) {
```
Comment (ColeBollig) [inline src/condor_utils/condor_event.cpp:462]: Is 24 valid for dt.tm_hour here?

Comment (johnkn) [reply src/condor_utils/condor_event.cpp:462]: I think so.  this line is from the old code.

Comment (johnkn) [reply src/condor_utils/condor_event.cpp:462]: in fact, from here down this is from the old code except for the last 2 lines.

**Review by ColeBollig — APPROVED** [review]
LGTM.

**Pull request merged**